tests/infer/mcmc/test_hmc.py

```diff
@@ -297,3 +297,23 @@ def potential_energy(params):
     posterior = torch.stack([sample["z"] for sample in posterior])
     assert_close(torch.mean(posterior), true_mean, rtol=0.05)
     assert_close(torch.std(posterior), true_std, rtol=0.05)
+
+
+@pytest.mark.parametrize('jit', [False, mark_jit(True)], ids=jit_idfn)
+@pytest.mark.parametrize('op', [torch.inverse, torch.cholesky])
+def test_singular_matrix_catch(jit, op):
+    def potential_energy(z):
+        return op(z['cov']).sum()
+
+    init_params = {'cov': torch.eye(3)}
+    potential_fn = potential_energy if not jit else torch.jit.trace(potential_energy, init_params)
+    hmc_kernel = HMC(potential_fn=potential_fn, adapt_step_size=False,
+                     num_steps=10, step_size=1e-20)
+    hmc_kernel.initial_params = init_params
+    hmc_kernel.setup(warmup_steps=0)
+    # setup an invalid cache to trigger singular error for torch.inverse
+    hmc_kernel._cache({'cov': torch.ones(3, 3)}, torch.tensor(0.), {'cov': torch.zeros(3, 3)})
```

Comment:
I don't know how to construct a potential_energy which triggers singular error for torch.inverse with an valid initial param. So I use this hack to setup an invalid initial param.